src/sortStudents.ts

```diff
@@ -1,16 +1,91 @@
 
 export interface Student {
   // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
-
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
   // write your function
+  const newStude = students.map((studier) => studier);
+  let result: Student[] = [];
+
+  function average(student: Student): number {
+    return student.grades.reduce((acc: number, val: number) => (acc + val),
+      0) / student.grades.length;
+  }
+
+  if (sortBy === 'grades') {
+    result = newStude.sort((a, b) => {
+      if (order === 'asc') {
+        return average(a) - average(b);
+      }
+
+      return average(b) - average(a);
+    });
```

Comment:
pls consider using ternaries
```suggestion
  order === 'asc' ? average(a) - average(b) : average(b) - average(a);
```